src/samples/public-client/UsernamePasswordFlow.java

```diff
@@ -66,20 +65,17 @@ private static void getAccessTokenFromUserCredentials() throws Exception {
 
             } catch (MalformedURLException e) {
                 e.printStackTrace();
+                throw new RuntimeException();
```

Comment:
Why? can't the original exception be re-thrown? (alternatively could this be made the inner exception)

Reply:
Without going too much into detail, you [can't throw checked exceptions from a lambda (or streams) without a lot more boilerplate code](https://stackoverflow.com/questions/27644361/how-can-i-throw-checked-exceptions-from-inside-java-8-streams). I didn't think it was worth it here. The simpler alternative is to just throw an unchecked exception.